raiden/utils/__init__.py

```diff
@@ -266,3 +266,20 @@ def wait_until(func, wait_for=None, sleep_for=0.5):
             res = func()
 
     return res
+
+
+def compare_versions(deployed_version, current_version):
```

Comment:
This function seems unnecessary, why not a plain equality check against both strings?

Reply:
Lexicographical order will probably work here, but I believe it's better to abstract the check away - do we have a versioning scheme set already? Are we sure that the version string won't change?

Reply:
My idea was to guarantee compatibility of the third version number (how do you call it actually? build?), and refuse to start if major or minor version are different. Minor version can be also backward compatible.

(Wondering if it makes sense with smart contracts)

Reply:
> My idea was to guarantee compatibility of the third version number

Yes, that sounds good. But isn't this simpler?

```python
deployed_version[:deployed_version.rfind('.')] == current_version[:current_version.rfind('.')]
```